### PR TITLE
Create OCP conditional branches rhel8 dockerfile

### PR DESCRIPTION
**Pull request by bentito**
Based on not the most popular answer here: https://stackoverflow.com/questions/43654656/dockerfile-if-else-condition-with-external-arguments

Yes it builds both branches, but this reads really cleanly to me, but also because I am not sure about the context if we try with a RUN build script.

### CODE REVIEW

File: Dockerfile.rhel8 [hunk 1]
```diff
@@ -79,3 +81,115 @@ LABEL io.k8s.display-name="OpenShift Hive" \
       io.k8s.description="This is an image used by operator-metering to to install and run Apache Hive." \
       io.openshift.tags="openshift" \
       maintainer="Chance Zibolski <czibolsk@redhat.com>"
```
Comment (timflannagan) [inline Dockerfile.rhel8]: Ah looks like this was missing in the last maintainer updates I made.

File: Dockerfile.rhel8 [hunk 2]
```diff
@@ -1,4 +1,6 @@
-FROM ubi8:8-released AS build
+ARG OPENSHIFT_CI
```
Comment (timflannagan) [inline Dockerfile.rhel8]: If we're just doing test runs, should we echo the value of this variable to make sure we're on the right track?

File: Dockerfile.rhel8 [hunk 3]
```diff
@@ -79,3 +81,115 @@ LABEL io.k8s.display-name="OpenShift Hive" \
...
+VOLUME /var/lib/hive
+
+USER 1002
+
+LABEL io.k8s.display-name="OpenShift Hive" \
+      io.k8s.description="This is an image used by operator-metering to to install and run Apache Hive." \
+      io.openshift.tags="openshift" \
+      maintainer="metering-team@redhat.com"
+
+FROM build-openshift-${OPENSHIFT_CI} AS final
```
Comment (timflannagan) [inline Dockerfile.rhel8]: This looks so strange to me 8-)

Comment (bentito) [reply Dockerfile.rhel8]: It is so strange. Agreed. I'll add an echo but it won't save much debug time, the build output seems to come all at once after awhile.

**Comment by timflannagan** [comment]
@bentito the PR for updating our CI to use the rhel8 dockerfile hasn't been merged yet, so this CI run won't expose much value for us. Should we just concede for now, merge that open PR, and attempt to make iterative changes as the feedback from failed CI runs is actually representative of what we're testing, or just update the dockerfile that CI is pointing to now (I assume the Dockerfile.okd).

**Comment by bentito** [comment]
> @bentito the PR for updating our CI to use the rhel8 dockerfile hasn't been merged yet, so this CI run won't expose much value for us. Should we just concede for now, merge that open PR, and attempt to make iterative changes as the feedback from failed CI runs is actually representative of what we're testing, or just update the dockerfile that CI is pointing to now (I assume the Dockerfile.okd).

Ah good point, the changes here aren't being hit.  I think we should try pointing CI at the Dockerfile.rhel8, I'll do a PR on release for that. Agree @timflannagan1 ?

**Comment by timflannagan** [comment]
> > @bentito the PR for updating our CI to use the rhel8 dockerfile hasn't been merged yet, so this CI run won't expose much value for us. Should we just concede for now, merge that open PR, and attempt to make iterative changes as the feedback from failed CI runs is actually representative of what we're testing, or just update the dockerfile that CI is pointing to now (I assume the Dockerfile.okd).
> 
> Ah good point, the changes here aren't being hit. I think we should try pointing CI at the Dockerfile.rhel8, I'll do a PR on release for that. Agree @timflannagan1 ?

As a workaround, we could just copy and paste what you have in this dockerfile to the dockerfile that CI is configured to use for now, before updating our release configuration to use the same dockerfile that ART is configured to use.

**Comment by bentito** [comment]
> > > @bentito the PR for updating our CI to use the rhel8 dockerfile hasn't been merged yet, so this CI run won't expose much value for us. Should we just concede for now, merge that open PR, and attempt to make iterative changes as the feedback from failed CI runs is actually representative of what we're testing, or just update the dockerfile that CI is pointing to now (I assume the Dockerfile.okd).
> > 
> > 
> > Ah good point, the changes here aren't being hit. I think we should try pointing CI at the Dockerfile.rhel8, I'll do a PR on release for that. Agree @timflannagan1 ?
> 
> As a workaround, we could just copy and paste what you have in this dockerfile to the dockerfile that CI is configured to use for now, before updating our release configuration to use the same dockerfile that ART is configured to use.

Okay, sounds faster.

**Comment by bentito** [comment]
@timflannagan1 yeah this approach will never work, assumes both build "branches" could work from CI, the ART one cannot, so errors out. Unless there was a way of accepting some build errors and continuing on...

**Comment by bentito** [comment]
> @timflannagan1 yeah this approach will never work, assumes both build "branches" could work from CI, the ART one cannot, so errors out. Unless there was a way of accepting some build errors and continuing on...

I don't think there is any way to ignore erroring on this line when in CI: ```FROM openshift/ose-metering-hadoop:latest```
@timflannagan1 Do you have a handle on doing this with a build script that could implement the conditional logic AND get the context right?

**Comment by timflannagan** [comment]
Superseded by #46 and #47

**Pull request closed without merging**